relay-event-normalization/src/timestamp.rs

```diff
@@ -39,15 +39,45 @@ impl Processor for TimestampProcessor {
             }
         }
 
+        event.process_child_values(self, state)?;
+
+        Ok(())
+    }
+
+    fn process_span(
+        &mut self,
+        span: &mut Span,
+        meta: &mut Meta,
+        _: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        if let Some(start_timestamp) = span.start_timestamp.value() {
+            if start_timestamp.into_inner().timestamp_millis() < 0 {
+                meta.add_error(Error::invalid(format!(
+                    "timestamp is too stale: {}",
+                    start_timestamp
+                )));
+                return Err(ProcessingAction::DeleteValueHard);
+            }
+        }
+        if let Some(end_timestamp) = span.timestamp.value() {
+            if end_timestamp.into_inner().timestamp_millis() < 0 {
+                meta.add_error(Error::invalid(format!(
+                    "timestamp is too stale: {}",
+                    end_timestamp
+                )));
+                return Err(ProcessingAction::DeleteValueHard);
+            }
+        }
```

Comment:
What if we put this logic in `process_timestamp` instead of `process_span`? That should automatically cover all timestamps, plus it would attach the meta error to the correct field.

Reply:
That will deal with only the timestamp itself, but we want to take action on the container of the timestamp. For spans, this means dropping the span; for transactions, this means dropping the entire event. We can't have this level of control in `process_timestamp` unfortunately.